lib/collections.go

```diff
@@ -270,293 +267,191 @@ type collectionsLib struct{}
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "list_flatten",
-				Unary:    flatten,
-			},
+
+		cel.Function("max",
+			cel.MemberOverload(
+				"list_max",
+				[]*cel.Type{listV},
+				typeV,
+				cel.UnaryBinding(max),
+			),
+			cel.Overload(
+				"max_list",
+				[]*cel.Type{listV},
+				typeV,
+				cel.UnaryBinding(max),
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "min_list",
-				Unary:    min,
-			},
-			&functions.Overload{
-				Operator: "list_min",
-				Unary:    min,
-			},
+
+		cel.Function("min",
+			cel.MemberOverload(
+				"list_min",
+				[]*cel.Type{listV},
+				typeV,
+				cel.UnaryBinding(min),
+			),
+			cel.Overload(
+				"min_list",
+				[]*cel.Type{listV},
+				typeV,
+				cel.UnaryBinding(min),
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "max_list",
-				Unary:    max,
-			},
-			&functions.Overload{
-				Operator: "list_max",
-				Unary:    max,
-			},
+
+		cel.Function("values",
+			cel.MemberOverload(
+				"map_values",
+				[]*cel.Type{mapKV},
+				listK,
+				cel.UnaryBinding(mapValues),
+			),
+			cel.Overload(
+				"values_map",
+				[]*cel.Type{mapKV},
+				listK,
+				cel.UnaryBinding(mapValues),
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "map_with_map",
-				Binary:   withAll,
-			},
-			&functions.Overload{
-				Operator: "map_with_update_map",
-				Binary:   withUpdate,
-			},
-			&functions.Overload{
-				Operator: "map_with_replace_map",
-				Binary:   withReplace,
-			},
+
+		cel.Function("with",
+			cel.MemberOverload(
+				"map_with_map",
+				[]*cel.Type{mapKV, mapKV},
+				mapKV,
+				cel.BinaryBinding(withAll),
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "zip_list",
-				Binary:   zipLists,
-			},
-			&functions.Overload{
-				Operator: "list_zip",
-				Binary:   zipLists,
-			},
+
+		cel.Function("with_update",
+			cel.MemberOverload(
+				"map_with_update_map",
+				[]*cel.Type{mapKV, mapKV},
+				mapKV,
+				cel.BinaryBinding(withUpdate),
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "map_keys",
-				Unary:    mapKeys,
-			},
-			&functions.Overload{
-				Operator: "keys_map",
-				Unary:    mapKeys,
-			},
+
+		cel.Function("with_replace",
+			cel.MemberOverload(
+				"map_with_replace_map",
+				[]*cel.Type{mapKV, mapKV},
+				mapKV,
+				cel.BinaryBinding(withReplace),
+			),
 		),
-		cel.Functions(
-			&functions.Overload{
-				Operator: "map_values",
-				Unary:    mapValues,
-			},
-			&functions.Overload{
-				Operator: "values_map",
-				Unary:    mapValues,
-			},
+
+		cel.Function("zip",
+			cel.MemberOverload(
+				"list_zip",
+				[]*cel.Type{listK, listV},
+				mapKV,
+				cel.BinaryBinding(zipLists),
+			),
+			cel.Overload(
+				"zip_list",
+				[]*cel.Type{listK, listV},
+				mapKV,
+				cel.BinaryBinding(zipLists),
+			),
 		),
 	}
 }
 
+func (collectionsLib) ProgramOptions() []cel.ProgramOption { return nil }
```

Comment:
I believe this function definition is still required and cannot be omitted?

Reply:
That's correct. This is required for Go interface satisfaction.